eventdata/parameter_sources/randomevent.py

```diff
@@ -278,7 +280,20 @@ def generate_event(self):
         # set host name
         event["hostname"] = "web-{}-{}.elastic.co".format(event["geoip_continent_code"],random.randrange(1,3))
 
+        # determine the raw event size (as if this were contained in nginx log file. We do not bother to
+        # reformat the timestamp as this is not worth the overhead.
+        if self.record_raw_event_size:
+            raw_event = '%s - - [%s] "%s %s HTTP/%s" %s %s "%s" "%s"' % (event["clientip"], event["@timestamp"],
+                                                                         event["verb"], event["request"],
+                                                                         event["httpversion"], event["response"],
+                                                                         event["bytes"], event["referrer"],
+                                                                         event["agent"])
+            raw_event_size = len(raw_event)
+        else:
+            raw_event_size = 0
```

Comment:
I was curious to see whether going through if/else in the existing cases (where `record_raw_event_size` isn't used) affects performance; using a quick `cProfile` based microbenchmark I observed a consistent 1 sec improvement moving `raw_event_size=0` before if and removing `else` with 200million iterations. I guess we can classify this as drop in the ocean?

Conversely when `record_raw_event_size is True` and `raw_event_size = 0` is defined before `if`, therefore reset inside if, it was ~1sec slower, with the same amount of iterations compared to this code.

Reply:
Thanks for double-checking. I'd also consider this a drop in the ocean. Also, the new approach works a bit different anyway (we have completely different code paths to create the document now in order to ensure we don't have any complex code on the hot code path).